### DISCUSSION TITLE
to run the test “iModelWriteTest”

### DISCUSSION THREAD

**Discussion by NguyenBentley** [iTwin/itwinjs-core#128] [Q&A] [answered]
I am studying iModelJs cloned from https://github.com/imodeljs/imodeljs.git and want to run the test “iModelWriteTest”

For this I added an entry in launch.json (the only change I add)
{
  "type": "node",
  "request": "launch",
  "name": "Run IModelWrite Test",
  "program": "${workspaceFolder}/core/backend/node_modules/@bentley/build-tools/scripts/test",
  "cwd": "${workspaceFolder}/core/backend",
  "args": [
	"--testDir=${workspaceFolder}/core/backend/lib/test/",
	"--inspect",
	"--recursive",
	"--grep",
	"IModelWriteTest"
  ],
  "port": 9229,
  "protocol": "inspector",
  "outputCapture": "std",
  "envFile": "${workspaceFolder}/.env",
},

the breakpoint is hit
Then while stepping in the file .\iModel\imodeljs\tools\oidc-signin-tool\src\TestUtility.ts, getAuthorizedClientRequestContext(),

i do not get any AccesToken then I get the message

Error: Browser is not downloaded. Run "npm install" or "yarn install"

Then I do “rush clean” then " rush install" then "rush build" 10x and debug but get the same error. Why ?

2. questions:

Do I need some credentials to run the mentioned test, because it is a integration test. When yes then please give us a dummy user & pw & …

Or tell us how to create a test user

Regard

**Comment by ramanujam-raman-zz** [comment, marked as answer]
These tests are currently not setup to be run as-is, and require a fair bit of configuration. You would have to setup a project with appropriate access to test users, and configurate the environment listed in tools/oidc-signin-tool/src/TestUsers.ts for these tests to work.

**Comment by NguyenBentley** [comment]
thank
could u say short about how to setup a imodeljs project with appropriate access to test users ? where can i find docs about it ? or some contact person ?
